README.md

```diff
@@ -32,3 +32,92 @@ To only generate the policies in the `helm` folder structure:
 ```bash
 make generate
 ```
+
+### Adding tests
+
+This repository uses the [app-build-suite](https://github.com/giantswarm/app-build-suite/) and the related testing setup.
+We have tried to make the test setup as simple as possible but some python knowledge is required.
+
+The tests use [python fixtures](https://docs.pytest.org/en/6.2.x/fixture.html) extensively to set up any resources we need in our tests.
+
+All fixtures can be found in [ensure.py](https://github.com/giantswarm/kyverno-policies/blob/main/helm/tests/ensure.py).
+Each fixtures should be structured in a similar way:
+```python
+@pytest.fixture
+def myresource(kubernetes_cluster):
+    # Yaml of whatever resource you want to create.
+    # The cluster_name variable is defined globally in ensure.py - so we always reuse the same names.
+    c = dedent(f"""
+        apiVersion: cluster.x-k8s.io/v1alpha3
+        kind: MyResource
+        metadata:
+          name: {cluster_name}
+        spec:
+          some:
+            spec: ""
+    """)
+
+    # Creating the resource for our test.
+    kubernetes_cluster.kubectl("apply", input=c, output=None)
+    LOGGER.info(f"MyResource {cluster_name} applied")
+
+    # Get the resource back from Kubernetes after it has been applied / defaulted.
+    raw = kubernetes_cluster.kubectl(
+        f"get MyResource {cluster_name}", output="yaml")
+
+    myresource = yaml.safe_load(raw)
+
+    # yield returns the object to our test case.
+    yield myresource
+
+    # Do cleanup after our testcase has ended.
+    kubernetes_cluster.kubectl(f"delete MyResource {cluster_name}", output=None)
+    LOGGER.info(f"MyResource {cluster_name} deleted")
+```
+
+The testcases can now look very simple as we only need to assert that the resources were created as we expected.
```

Comment:
Above you're creating a fixture for a `MyResource` but here you're using CAPI CRs. It might help to use `myresource` in this test (if I'm understanding correctly).

Reply:
I might change the sample above to have `awscluster` instead. Maybe that would make it less confusing :D

Reply:
That sounds better. Otherwise they feel disconnected.

Reply:
PTAL